app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -141,6 +144,52 @@ protected void onCreate(Bundle savedInstanceState) {
         mGestureDetector = new GestureDetector(this, this);
     }
 
+    protected void requestMajorLayoutUpdates() {
+        final View decorView = getWindow().getDecorView();
+        decorView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+
+            int mPreviousDecorViewFrameHeight = 0;
+
+            @Override
+            public void onGlobalLayout() {

```

Comment:
Hey,

Yeah, the idea is basically that you call this to request an update any time your size changes dramatically. 

It's not necessarily specific to keyboard popup, and I don't think we would suppress the event if the keyboard hadn't changed. The idea is essentially that if you have any layout code that is dependent on screen size, this will give you the right hook for when to re-evaluate your static screen size.

I can add docs